src/R2N.jl

```diff
@@ -0,0 +1,387 @@
+using LinearAlgebra
+using LinearOperators
+export R2N, R2NSolver
+export ShiftedLBFGSSolver
+
+abstract type AbstractShiftedLBFGSSolver end
+
+struct ShiftedLBFGSSolver <: AbstractShiftedLBFGSSolver
+  # Shifted LBFGS-specific fields
+end
+
+"""
+    R2N(nlp; kwargs...)
+
+A first-order quadratic regularization method for unconstrained optimization.
```

Comment:
Same as FOMO?

Reply:
I think I need to remove the first-order as we have Hessian or hessian operator in there, right?